src/c_api/symbolize.rs

```diff
@@ -319,42 +277,44 @@ unsafe fn from_cstr(cstr: *const c_char) -> PathBuf {
 }
 
 
-/// Create an instance of blazesym a symbolizer for C API.
+/// Options for configuring `blaze_symbolizer` objects.
+#[repr(C)]
+#[derive(Debug)]
+pub struct blaze_symbolizer_opts {
+    /// Whether to gather line number information.
+    pub line_numbers: bool,
+    /// Whether to use debug symbols.
+    pub debug_info: bool,
```

Comment:
I think our original naming was confusing. `line_numbers` also implies debug info usage, so generic `debug_info` name makes things weird.

I think we generally have two sets of expensive features that require DWARF:
  - source code location (file + line numbers);
  - inline functions symbolization;

For former, something more generic than "line numbers " name would probably be good. `src_location` or something like that?

For the latter more specific `inline_funcs` or something like that, maybe?

Reply:
Wondering some more, could it be that "debug_info" just means to use DWARF func symbols info for symbolization, but not inline funcs? E.g., if ELF is stripped of symbols, but DWARF is present?

But if so, that seems like a third option, something like `use_debug_symbols` or something like that. WDYT?

Reply:
So for this pull request the intention pretty much was to just rework the way we provide features. I.e., it's really just a repackaging. I was going to look closer at the need for line number logic and everything else.

I'd be fine with dropping `line_numbers` and just keeping `debug_info`. Can rename to `use_debug_symbols` or whatever. What we need to clarify, I think, is whether that constitutes an exclusive option or a fallback. Currently, I believe, a regular ELF symbol lookup is performed first and only if we don't find anything we check DWARF (it changes a bit with line numbers, but that's the gist I recall).

What makes inline functions expensive? Isn't it just a matter of reporting more symbols for one address? All our APIs already can do that.

Reply:
I think I'd do it the other way. From user perspective what I care about is whether I get stack traces with source code location information (currently `line_numbers` but it's a bit wrong name), and, completely separately  and independently from that, whether my stack traces include inline functions or not. Both of these options should imply that we need DWARF.

I'm actually not yet sure how valuable is just saying "use DWARF for symbol names instead of ELF symbols", so I'd rather initially drop that option instead. We might want to add it later, but this isn't something that was super useful in my experience so far.

As for what makes inline functions more expensive. I think it's more DWARF info that needs to be pre-processed and cached. So more memory, mainly, but also more CPU. So you can't always afford this, especially for huge production binaries. So this has to be tunable.

Line numbers are expensive as well, also both CPU and memory-wise, but its data is coming from different sections of DWARF. So there isn't anything to even reuse between file/line info and inline functions.

Both very useful features. Both expensive :)